hive_agent/wallet/wallet.py

```diff
@@ -1,27 +1,41 @@
+import logging
+import sys
+from datetime import datetime
+
 from typing import Any, Dict, List
 
 from eth_account import Account
 from eth_account.datastructures import SignedMessage
 from eth_account.signers.local import LocalAccount
 from eth_account.messages import encode_defunct
 
+logging.basicConfig(stream=sys.stdout, level=logging.INFO)
+logging.getLogger().addHandler(logging.StreamHandler(stream=sys.stdout))
+
 
 class ImmutableDict:
     def __init__(self):
         self._dict = {}
-
+        logging.info("created new ImmutableDict")
+        
     def add(self, key, value):
         if key in self._dict:
+            logging.error(f"key '{key}' already exists. Cannot add the value.")
             raise KeyError(f"key '{key}' already exists. Cannot update the value.")
         self._dict[key] = value
+        logging.info(f"Added key: {key}")
 
     def get(self, key):
-        return self._dict.get(key, None)
+        value = self._dict.get(key, None)
+        logging.info(f"Retrieved key: {key} with value: {value}")
```

Comment:
use `logging.debug` here as well and the subsequent functions (except the error handling logs)